bolt-modules/boltlib/lib/puppet/functions/run_command.rb

```diff
@@ -56,6 +57,22 @@ def run_command_with_description(command, targets, description = nil, options =
     options = options.transform_keys { |k| k.sub(/^_/, '').to_sym }
     options[:description] = description if description
 
+    # Ensure env_vars is a hash and that each hash value is transformed to JSON
+    # so we don't accidentally pass Ruby-style data to the target.
+    if options[:env_vars]
+      unless options[:env_vars].is_a?(Hash)
+        raise Bolt::ValidationError, "Option 'env_vars' must be a hash"
+      end
+
+      unless options[:env_vars].keys.all? { |k| k.is_a?(String) }
+        raise Bolt::ValidationError, "Keys for option 'env_vars' must be strings"
```

Comment:
Hm, can you add the specific value(s) that are failing here? I know other validation errors here don't have identifying information, and we should probably fix those too - but in lieu of line numbers (in YAML plans) it'd be handy to have the value that's failing so users can find it.